packages/context/src/interceptor-chain.ts

```diff
@@ -79,19 +90,22 @@ class InterceptorChainState<C extends Context = Context> {
  *
  * @typeParam C - `Context` class or a subclass of `Context`
  */
-export class GenericInterceptorChain<C extends Context = Context> {
+export class GenericInterceptorChain<
+  C extends Context = Context,
+  T = InvocationResult
```

Comment:
Ditto, what are the arguments for providing default values for `C` and `T`?

Let's rename `T` to `R`.